concurrent_test.go

```diff
@@ -511,6 +537,32 @@ func (rs historyRecords) Swap(i, j int) {
 	rs[i], rs[j] = rs[j], rs[i]
 }
 
+func validateIncrementalTxid(rs historyRecords, op OperationType) error {
+	if len(rs) == 0 {
+		return nil
+	}
+
+	lastTxid := rs[0].Txid
+
+	if op == Write {
+		for i := 1; i < len(rs); i++ {
+			if rs[i].Txid <= lastTxid {
+				return fmt.Errorf("detected non-incremental txid(%d, %d) in write mode", lastTxid, rs[i].Txid)
+			}
+			lastTxid = rs[i].Txid
+		}
+	} else {
```

Comment:
There is a bit of duplication here, just an idea but we could use one loop and be a bit more conditional, something like:

```go
    for i := 1; i < len(rs); i++ {
        if (op == Write && rs[i].Txid <= lastTxid) || (op == Read && rs[i].Txid < lastTxid) {
            return fmt.Errorf("detected non-incremental txid(%d, %d) in %s mode", lastTxid, rs[i].Txid, op)
        }
        lastTxid = rs[i].Txid
    }
```

Feel free to ignore if the explicit goal is to be very verbose here.

Reply:
Yes, I intentionally duplicate the code, because it has a little better performance.  But I agree that readability outweigh here, and the performance decreasing might be negligible.

Updated & thx